applications/StructuralMechanicsApplication/tests/cpp_tests/test_yield_surfaces_3d.cpp

```diff
@@ -0,0 +1,244 @@
+            KRATOS_CHECK_NEAR(Tres[comp],  TestT[comp],  1.0e-3);
+        }
+    }
+
+    /** 
+    * Check the correct calculation of the initial threshold of the yield surfaces
+    */
+    KRATOS_TEST_CASE_IN_SUITE(YieldSurfacesInitialUniaxialThreshold, KratosStructuralMechanicsFastSuite)
+    {
+        Properties rMaterialProperties;
+        Vector Strain, Stress;
+        GenerateTestVariables(Stress, Strain, rMaterialProperties);
+
+        // Analytical solutions of the initial threshold
+        double MCres, VMres, DPres, Rres, Tres, SJres;
+        MCres = 30.0e6;
+        VMres = 3.0e6;
+        DPres = 7.50918e+06;
+        Rres  = 3.0e6;
+        Tres  = 3.0e6;
+        SJres = 65.4654;
+
+        double TestMC = 0.0, TestVM = 0.0, TestDP = 0.0, TestR = 0.0, TestT, TestSJ = 0.0;
+        MC::GetInitialUniaxialThreshold(rMaterialProperties, TestMC);
+        VM::GetInitialUniaxialThreshold(rMaterialProperties, TestVM);
+        DP::GetInitialUniaxialThreshold(rMaterialProperties, TestDP);
+        R::GetInitialUniaxialThreshold(rMaterialProperties, TestR);
+        T::GetInitialUniaxialThreshold(rMaterialProperties, TestT);
+        SJ::GetInitialUniaxialThreshold(rMaterialProperties, TestSJ);
+
+        // Check the results!
+        KRATOS_CHECK_NEAR(MCres, TestMC, 1.0e-3);
+        KRATOS_CHECK_NEAR(VMres, TestVM, 1.0e-3);
+        KRATOS_CHECK_NEAR(DPres, TestDP, 1.0e-3);
+        KRATOS_CHECK_NEAR(Rres, TestR, 1.0e-3);
+        KRATOS_CHECK_NEAR(Tres, TestT, 1.0e-3);
+        KRATOS_CHECK_NEAR(SJres, TestSJ, 1.0e-3);
+    }
+
+    /** 
+    * Check the correct calculation of the Damage Parameter of the yield surfaces
+    */
+    KRATOS_TEST_CASE_IN_SUITE(YieldSurfacesIDamageParameter, KratosStructuralMechanicsFastSuite)
+    {
+        Properties rMaterialProperties;
+        Vector Strain, Stress;
+        GenerateTestVariables(Stress, Strain, rMaterialProperties);
+        double characteristic_length = 0.1;
+
+        // Analytical solutions damage parameter Linear Softening
+        double MCres, VMres, DPres, Rres, Tres, SJres;
+        MCres = -0.00214286;
+        VMres = -0.214286;
+        DPres = -0.00214286;
+        Rres  = -0.214286;
+        Tres  = -0.00214286;
+        SJres = -4.5e+08;
+
+        double TestMC = 0.0, TestVM = 0.0, TestDP = 0.0, TestR = 0.0, TestT = 0.0, TestSJ = 0.0;
+        MC::CalculateDamageParameter(rMaterialProperties, TestMC, characteristic_length);
+        VM::CalculateDamageParameter(rMaterialProperties, TestMC, characteristic_length);
+        DP::CalculateDamageParameter(rMaterialProperties, TestMC, characteristic_length);
+        R::CalculateDamageParameter(rMaterialProperties, TestMC, characteristic_length);
+        T::CalculateDamageParameter(rMaterialProperties, TestMC, characteristic_length);
+        SJ::CalculateDamageParameter(rMaterialProperties, TestMC, characteristic_length);
+
+        // Check the results!
+        KRATOS_CHECK_NEAR(MCres, TestMC, 1.0e-4);
```

Comment:
Why such a low tolerance?

Reply:
this is a good question... I reviewed this with @RiccardoRossi . Which one could be reasonable?

Thank you